bug: the TestMulticastPerNode is not robust

```
--- FAIL: TestMulticastPerNode (0.01s)
    --- FAIL: TestMulticastPerNode/MulticastPerNodeNoSendWaiting/Servers=9/IgnoredNodes=[] (0.00s)
        oneway_test.go:216: MulticastPerNodeNoSendWaiting -> 8, expected 5, nodeID=3, ignore=false
    --- FAIL: TestMulticastPerNode/MulticastPerNodeNoSendWaitingIgnoreNodes/Servers=3/IgnoredNodes=[0_1] (0.00s)
        oneway_test.go:216: MulticastPerNodeNoSendWaitingIgnoreNodes -> 18, expected 20, nodeID=3, ignore=false
    --- FAIL: TestMulticastPerNode/MulticastPerNodeSendWaitingIgnoreNodes/Servers=3/IgnoredNodes=[0] (0.00s)
        oneway_test.go:216: MulticastPerNodeSendWaitingIgnoreNodes -> 17, expected 15, nodeID=1, ignore=false
FAIL
```
The above issue is not related to a prior issue raised by deepsource, but we should fix those issues anyway.

--- a/internal/tests/oneway/oneway_test.go
+++ b/internal/tests/oneway/oneway_test.go
@@ -3,6 +3,7 @@ package oneway_test
 import (
 	context "context"
 	"fmt"
+	"slices"
 	"sync"
 	"testing"
 
@@ -125,15 +126,8 @@ func TestMulticastPerNode(t *testing.T) {
 	add := func(n uint64, id uint32) uint64 { return n + uint64(id) }
 
 	// makeIgnoreFunc creates a function that checks if a node ID should be ignored.
-	// It captures its own copy of the ignoreNodes slice to avoid data races.
 	makeIgnoreFunc := func(ignoreNodes []uint32) func(uint32) bool {
-		ignoreSet := make(map[uint32]bool, len(ignoreNodes))
-		for _, id := range ignoreNodes {
-			ignoreSet[id] = true
-		}
-		return func(id uint32) bool {
-			return ignoreSet[id]
-		}
+		return func(id uint32) bool { return slices.Contains(ignoreNodes, id) }
 	}
 
 	// transformation function that uses the MapRequest interceptor

